src/transactions.rs

```diff
@@ -109,92 +138,94 @@ pub trait RevaultTransaction: fmt::Debug + Clone + PartialEq {
         pubkey: BitcoinPubKey,
         signature: BitcoinSig,
     ) -> Result<Option<Vec<u8>>, Error> {
-        if let Some(ref mut psbtin) = self.inner_tx_mut().inputs.get_mut(input_index) {
-            // BIP174:
-            // For a Signer to only produce valid signatures for what it expects to sign, it must
-            // check that the following conditions are true:
-            // -- If a witness UTXO is provided, no non-witness signature may be created.
-            let prev_txo = psbtin.witness_utxo.as_ref().ok_or_else(|| {
-                debug_assert!(
-                    false,
-                    "Cannot be reached. We only create transactions with witness_utxo."
-                );
-                Error::InputSatisfaction(format!(
-                    "No previous witness txo for psbtin: '{:?}'",
-                    psbtin
-                ))
-            })?;
-            if psbtin.non_witness_utxo.is_some() {
-                debug_assert!(
-                    false,
-                    "Cannot be reached. We never create transactions with non_witness_utxo."
-                );
+        let psbtin = match self.inner_tx_mut().inputs.get_mut(input_index) {
```

Comment:
use again ok_or 
```suggestion
        let psbtin = match self.inner_tx_mut().inputs.get_mut(input_index).ok_or(
           Error::InputStatisfaction(...
        )?;
```

Reply:
Just made it an assert!() right now as i'm going to tackle deserialization checks ASAP

Reply:
Just made it an assert!() right now as i'm going to tackle deserialization checks ASAP